Library/Homebrew/utils/analytics.rb

```diff
@@ -69,12 +70,53 @@ def report(type, metadata = {})
         nil
       end
 
+      def report_influx(category, metadata = {})
+        curl = Utils::Curl.curl_executable
+        args = []
+        tags = []
+
+        metadata[:version] = HOMEBREW_VERSION
+        metadata[:product] = HOMEBREW_PRODUCT
+        metadata.each do |key, value|
+          next unless key
+          next unless value
+
+          tags << "#{key}=#{value.sub(" ", "\\ ")}"
+        end
+
+        args += %W[
+          --max-time 3
+          --header "Content-Type: text/plain; charset=utf-8"
+          --header "Accept: application/json"
+          --header "Authorization: Token #{TOKEN}"
+          --data-raw '#{category},#{tags.join(",")} count=1i #{Time.now.to_i}'
+        ]
```

Comment:
```suggestion
        args += [
          "--max-time", "3",
          "--header", "Content-Type: text/plain; charset=utf-8",
          "--header", "Accept: application/json",
          "--header", "Authorization: Token #{TOKEN}",
          "--data-raw", "'#{category},#{tags.join(",")} count=1i #{Time.now.to_i}'"
        ]
```

I believe that this is the source of the "unauthorized" problem for me. Since `%W[]` just splits on any whitespace, it ignores the quotation marks around the headers. So `args` actually is:

```ruby
["--max-time", "3", "--header", '"Content-Type:', "test/plain;", 'charset=utf-8"', "--header", '"Accept:', 'application/json"', "--header", '"Authorization:', "Token", '#{TOKEN}"', ...]
```

After making this change, I no longer get the unauthorized error. Now, though, nothing is printed but I think this is because the call doesn't actually display anything. When I run it directly, I get nothing printed. In both cases, I get a `204` response code [implying that the request was accepted](https://docs.influxdata.com/influxdb/cloud/api/#tag/Response-codes).

Reply:
Thanks!